cmd/cluster-agent/subcommands/start/command.go

```diff
@@ -198,17 +202,15 @@ func start(log log.Component, config config.Component, taggerComp tagger.Compone
 
 	// Initialize remote configuration
 	var rcClient *rcclient.Client
-	var rcService *rcservice.Service
-	if pkgconfig.IsRemoteConfigEnabled(pkgconfig.Datadog) {
+	rcService, isSet := rcServiceOptional.Get()
+	if pkgconfig.IsRemoteConfigEnabled(pkgconfig.Datadog) && isSet {
```

Comment:
While you're here, this let's us remove a direct reference to the global config object, in favor the config component:

```suggestion
	if pkgconfig.IsRemoteConfigEnabled(config) && isSet {
```